addons/misra.py

```diff
@@ -1628,6 +1638,35 @@ def misra_18_5(self, data):
                 self.reportError(var.nameToken, 18, 5)
 
 
+    def misra_18_7(self, data):
+        structScopes = []
+        for token in data.tokenlist:
+            if not isStruct(token):
+                continue
+            structScopes.append(token.valueType.typeScope)
+
+        for scope in structScopes:
+            token = scope.bodyStart.next
+            while token != scope.bodyEnd and token is not None:
+                # Handle nested structures to not duplicate an error.
+                if token.str == 'struct':
```

Comment:
does this work?

    if token.str == '{':
        token = token.link

Reply:
Actually, no. I nedd a struct token to get its scope using `isStruct`. This is necessary because I want to iterate over this scope ignoring possible nested structs like this:
```c
struct {
  struct {
    struct {
      uint8_t a3[42];
    } s3;
  } s2;
} s1;
```

Reply:
yes with my code you would ignore all nested scopes.